op-bindings/bindgen/etherscan_contracts.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"text/template"
+)
+
+type etherscanContract struct {
+	Name             string
+	Abi              string
+	DeployedBytecode string
+	InitBytecode     string
+	DeployedAddress  string
+	DeploymentTxHash string
+	DeployerAddress  string
+	ExpectedSalt     string
+}
+
+type etherscanContracts struct {
+	Verified struct {
+		StandardDeployment []etherscanContract `json:"standardDeployment"`
+		ProxyDeployment    []etherscanContract `json:"proxyDeployment"`
+	}
+	Unverified []etherscanContract `json:"unverified"`
+	Immutables struct {
+		ProxyDeployment []etherscanContract `json:"proxyDeployment"`
+	}
+}
+
+type etherscanApiResponse struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Result  interface{} `json:"result"`
+}
+
+type etherscanRpcResponse struct {
+	JsonRpc string      `json:"jsonrpc"`
+	Id      int         `json:"id"`
+	Result  interface{} `json:"result"`
+}
+
+type txInfo struct {
+	Input string `json:"input"`
+}
+
+type contractMetadata struct {
+	Name            string
+	Package         string
+	DeployedBin     string
+	InitBin         string
+	DeploymentSalt  string
+	DeployerAddress string
+}
+
+const (
```

Comment:
This is fine to use hardcoded strings for the urls since you have it already implemented but another way to do it would be to encapsulate the strings into a client like so:

```go
key := "my-api-key"
client, err := NewEtherscanClient("https://api.etherscan.io", key)
abi, err := client.GetAbi(address)
tx, err := client.GetTransaction(hash)
```

Then this client is reusable across networks and can be tested independently

Reply:
I've refactored the CLI to use the client approach